01-node-tutorial/answers/prompter.js

```diff
@@ -1,54 +1,122 @@
-let item = "Enter something below.";
+let textLabel = 'Please enter sample text:';
+let text = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit...';
+let bgColorLabel = "Please choose background color:";
+let bgColor = 'DodgerBlue';
+let colorLabel = 'Please choose text color:';
+let color = 'white';
+
+const getRandomInt = (min, max) => {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+const getRandomColor = () => {
+  return colors[getRandomInt(0, colors.length - 1)];
+}
+
+const getTenRandomColors = () => {
+  const colors = new Set();
+  let color;
+  while (colors.size < 10) {
+    color = getRandomColor();
+    // color = `color${colors.size + 1}`
+    colors.add(color);
+  }
+  return Array.from(colors);
+}
 
 // here, you can change the form below to modify the input fields and what is displayed.
 // This is just ordinary html with string interpolation.
 const form = () => {
+  const colors1 = getTenRandomColors();
+  const options1 = colors1.map((color) => `<option value="${color}" style="background-color: ${color} !important;" onMouseOver="this.style.backgroundColor='${color}'">${color}</option>`).join('');
+  const colors2 = getTenRandomColors();
+  const options2 = colors2.map((color) => `<option value="${color}" style="background-color: ${color};">${color}</option>`).join('');
   return `
-  <body>
-  <p>${item}</p>
-  <form method="POST">
-  <input name="item"></input>
+  <body style="background-color: ${bgColor}; color: ${color}; font-size: 20px; display: flex; flex-direction: column; align-items: center;">
+  <h1>Test Colors</h1>
+  <div>Sample text: ${text}</div>
+  <div>Background color: ${bgColor}</div>
+  <div>Text color: ${color}</div>
+  <br>
+  <form method="POST" style="display: flex; flex-direction: column; align-items: center; row-gap: 10px;">
+  <div>
+    <label for="text" style="margin-right: 5px;">${textLabel}</label>
+    <input name="text" id="text" required></input>
+  </div>
+  <div>
+    <label for="bgColor" style="margin-right: 5px;">${bgColorLabel}</label>
+    <select name="bgColor" id="bgColor" style="width: 200px; background-color: ${bgColor};" required onchange="updateSelectColor(this)">
+    <option disabled selected value> -- select an option -- </option>
+      ${options1}
+    </select>
+  </div>
+  <div>
+    <label for="color" style="margin-right: 5px;">${colorLabel}</label>
+    <select name="color" id="color" style="width: 200px;  background-color: ${color};" required onchange="updateSelectColor(this)">
+    <option disabled selected value> -- select an option -- </option>
+      ${options2}
+    </select>
+  </div>
+  <br>
   <button type="submit">Submit</button>
   </form>
   </body>
+  <script>
+    function updateSelectColor(selectElement) {
+      const selectedOption = selectElement.options[selectElement.selectedIndex];
+      selectElement.style.backgroundColor = selectedOption.style.backgroundColor;
+    }
+  </script>
   `;
 };
```

Comment:
Can we let the HTML out of the JS at this moment
nice work done but let's just focus on the js part here so avoid rendering too much template, or maybe you can render the template outside the server file